hooks/useCustomStyle.tsx

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useState } from "react";
+
+function isMobile() {
+    return globalThis.matchMedia("(max-width: 767px)").matches
+}
+
+function getOneSidePadding(mainElement: HTMLElement) {
+
+    return Number(getComputedStyle(mainElement).paddingLeft.replace("px", "")) ;
+}
+
+function getPseudoElementWidth(currentElement: HTMLElement, pseudoName : string) {
+
+    return Number(getComputedStyle(currentElement, pseudoName).width.replace('px', ''))
+}
+
+function isInCodeContainer(currentElement : HTMLElement) {
+    const containerElement = currentElement.closest('.code-container')!
+    return containerElement?.scrollLeft
+}
+
+function getValuesForCSSLeft( currentElement : HTMLElement) {
+
+    const screenWidth = globalThis.window.outerWidth;
+    const mainElement = document.querySelector('main')!
+    const mainLeftPadding = getOneSidePadding(mainElement)
+
+    const allowedContentFullWidth = screenWidth - (mainLeftPadding * 2)
+
+    const elementXPos = currentElement.getBoundingClientRect().x - mainLeftPadding
+    const elementXPosRatio = elementXPos / allowedContentFullWidth
+
+    return {
+        elementXPos,
+        allowedContentFullWidth,
+        elementXPosRatio,
+    }
```

Comment:
can you please explain to me the logic in this function? I don't quite understand why the window and  the`<main/>` are related to the calculations required... 🤔

Reply:
![KakaoTalk_20230421_190809795](https://user-images.githubusercontent.com/7471674/233610019-caf79599-9d9a-4c55-a9ab-c3e7fa0f80c8.jpg)